src/query-builder.ts

```diff
@@ -1,161 +1,160 @@
-import type { FieldsArgs, FilterValue, QueryIO, QueryInput } from './types'
+import type { EntityQueryMap, FieldsArgs, FilterValue, QueryInput } from './types'
 
 /**
- * Builder class for constructing GraphQL queries
+ * Converts a JSON object to a GraphQL arguments string
+ * Handles enum values correctly (removes quotes from values that appear to be enums)
+ *
+ * This method is needed for complex nested objects like 'where' filters
+ * @param obj - The object to convert to GraphQL arguments
+ * @returns A GraphQL arguments string
  */
-export class QueryBuilder {
+const jsonToGraphQLArgs = (obj: any): string => {
```

Comment:
is there a reason to use arrow functions here? In general its better to use declared functions for better error tracking